typescript/introduction/tsc-basic-types/tsc-object-type-comparison.md

```diff
@@ -0,0 +1,173 @@
+---
+author: nem035
+type: normal
+category: must-know
+aspects:
+  - workout
+  - deep
+links:
+  - '[object Type](https://www.typescriptlang.org/docs/handbook/basic-types.html#object){documentation}'
+  - '[Play with object types](https://www.typescriptlang.org/play/index.html#code/MYewdgzgLgBAhgLhgeQEYCsCmxYF4YDeMmYA1gJZJQBOArpjAL4DcAUK6JLKkiBtnkLEylGDXpM2HcNBjAkREhSSoQIADaY4YJjHyKRVOgxZA){website}'
+
+---
+
+# `Object` vs `object` vs `{}`
+
+---
+## Content
+
+TypeScript has 3 ways to represent an object with slightly different behaviors.
+
+> 💡 An object can be thought of as a collection of name value pairs
+
+### `Object`
+
+Since any defined value in JavaScript is an object[1], the `Object` type can be used to represent any type that is considered to contain a value:
+
+```ts
+let num: Object = 3;
+let str: Object = 'enki';
+let arr: Object = [1, 2, 3];
+// ...
+// only null and undefined
+//  give an error
+let n: Object = null; // error
+let u: Object = undefined; // error
+```
+
+This often leads to confusion.
+
+Usually by "object" we only mean collections of named values (e.g. maps, sets, arrays, plain objects), but not primitive values[2] (e.g. numbers or strings).
+
+### `object`
+
+The recommended type to represent an object is `object`.
```

Comment:
```suggestion
The recommended type to represent an object is `object` with any shape.
```

Reply:
or with an arbitrary shape

Reply:
Flipped this whole insight around, good suggestions 👍